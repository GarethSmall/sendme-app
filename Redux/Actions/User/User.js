// @flow

import { ActionType } from '../../../SharedTypes/ActionType';

export function login({ username, password } : { username : String, password : String }) {
  if (username === 'G' && password === '1') {
    return {
      type: ActionType.USER.LOGIN_SUCCESS,
    };
  } else {
    return {
      type: ActionType.USER.LOGIN_ERROR,
      response: {
        message: 'Inccorect Username/Password',
      },
    };
  }
}
