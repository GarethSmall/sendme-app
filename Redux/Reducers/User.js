// @flow
import type { UserState } from "../../SharedTypes/Reducer/User";
import type { UserAction, UserLoginError } from "../../SharedTypes/Actions/User";
import { ReducerType } from "../../SharedTypes/ReducerType";

export const initialState: UserState = {
  isLoggedIn: false,
  loginError: null
};

export default function(state : UserState = initialState, action : UserAction) : UserState {
  switch (action.type) {
    case ReducerType.USER.LOGIN_SUCCESS: {
      return {
        ...state,
        isLoggedIn: true,
        loginError: null
      };
    }
    case ReducerType.USER.LOGIN_ERROR: {
      const a = ((action : any) : UserLoginError);
      return {
        ...state,
        loginError: a.response.message
      };
    }
    default: {
      return state || initialState;
    }
  }
}
