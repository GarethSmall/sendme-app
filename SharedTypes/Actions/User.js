// @flow

export type UserLoginSuccess = {
  type : string,
}
export type UserLoginError = {
  type : string,
  response : {
    message : string,
  }
}

export type UserAction =
  | UserLoginSuccess
  | UserLoginError;