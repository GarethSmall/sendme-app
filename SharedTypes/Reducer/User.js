export type UserState = {
  isLoggedIn : boolean,
  loginError : ?string,
}