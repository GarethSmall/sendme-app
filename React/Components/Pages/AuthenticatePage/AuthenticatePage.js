import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { login } from '../../../../Redux/Actions/User/User';
import { bindActionCreators } from 'redux';
import type { Store } from '../../../../SharedTypes/Store';
import Login from '../Login/Login';

const actionsToBind = { login };
type ActionProps = { actions : typeof actionsToBind };

type FieldProps = {
  +isLoggedIn : boolean,
};

const mapActionsToProps = (dispatch) : ActionProps => ({ actions: bindActionCreators(actionsToBind, dispatch) });

const mapStateToProps = (state : Store) : FieldProps => ({
  isLoggedIn: state.User.isLoggedIn,
});

type Props = ActionProps;

@connect(mapStateToProps, mapActionsToProps)
export default class AuthenticatePage extends PureComponent<Props> {

  render() {
    const { isLoggedIn } = this.props;
    return isLoggedIn ? (
      <h1>Yahoo you're logged in!</h1>
    ) : (
      <Login />
    );
  }
}