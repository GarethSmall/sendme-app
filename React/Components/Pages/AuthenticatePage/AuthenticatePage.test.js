// We'll need to import react to use react component syntax
import React from 'react';
// We'll need to import Enzyme to configure our adapter and shallow to shallowly mount are component
import Enzyme, { shallow } from 'enzyme';
// We'll need enzyme react adapter to use enzyme to test react
import Adapter from 'enzyme-adapter-react-16';
import AuthenticatePage from './AuthenticatePage';
// import json conversion function
import toJSON from 'enzyme-to-json';
// import our initial user state
import { initialState as userInitialState } from '../../../../Redux/Reducers/User';
// import our mock store
import configureStore from 'redux-mock-store';
// Set enzymes adapter to enzyme-adapter-react-16
Enzyme.configure({ adapter: new Adapter() });

// create our mock store
const mockStore = configureStore([]);
// create our initial state
const initialState = {
  User: {
    userInitialState,
  },
};

// describe our test suite
describe('AuthenticatePage', () => {
  //define a test case
  test('should render logged out', () => {
    // mount component
    const component = shallow(<AuthenticatePage />, { context: { store: mockStore(initialState) } });
    // convert to JSON tree
    const tree = toJSON(component);
    // test that our component matches the snapshot
    expect(tree).toMatchSnapshot();
  });
  test('should render logged in', () => {
    initialState.User.isLoggedIn = true;
    const component = shallow(<AuthenticatePage />, { context: { store: mockStore(initialState) } });
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
  });
});
