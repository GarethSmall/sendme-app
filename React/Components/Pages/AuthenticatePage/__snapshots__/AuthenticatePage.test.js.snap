// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatePage should render logged in 1`] = `
<AuthenticatePage
  actions={
    Object {
      "login": [Function],
    }
  }
  isLoggedIn={true}
/>
`;

exports[`AuthenticatePage should render logged out 1`] = `
<AuthenticatePage
  actions={
    Object {
      "login": [Function],
    }
  }
/>
`;
